File: HW1/VB/vbnethw1/Form1.vb
Imports System.Runtime.CompilerServices

Public Class Form1
    Dim rnd As New Random
    Dim pari As Boolean = True


    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        For i = 0 To 10
            PictureBox1.BackColor = Color.FromArgb(255, rnd.Next(255), rnd.Next(255), rnd.Next(255))
            Button1.Text = 10 - i
            Threading.Thread.Sleep(1000)
            Refresh()
        Next
    End Sub

    Private Sub Button3_Click(sender As Object, e As EventArgs) Handles Button3.Click
        If pari Then
            PictureBox1.Location = New Point(400, 162)
            pari = False

        Else
            PictureBox1.Location = New Point(39, 162)
            pari = True

        End If

    End Sub

    Private Sub PictureBox1_Click(sender As Object, e As EventArgs) Handles PictureBox1.Click

    End Sub

    Private Sub button2Hover(sender As Object, e As EventArgs) Handles Button2.MouseHover
        Button1.BackColor = Color.FromArgb(255, rnd.Next(255), rnd.Next(255), rnd.Next(255))
        Button2.BackColor = Color.FromArgb(255, rnd.Next(255), rnd.Next(255), rnd.Next(255))
        Button3.BackColor = Color.FromArgb(255, rnd.Next(255), rnd.Next(255), rnd.Next(255))
    End Sub
End Class

File: Homework1/Homework1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Homework1
{
    public partial class Form1 : Form
    {
        Boolean pari = true;
        private Random r = new Random();

        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {

            for (int i = 0; i < 10; i++)
            {
                pictureBox1.BackColor = Color.FromArgb(r.Next(0, 255),r.Next(0, 255), r.Next(0, 255));
                System.Threading.Thread.Sleep(1000);
                this.effect1.Text = (10-i).ToString();
                this.Refresh();
            }
            this.effect1.Text = "Effect 1";
        }

        private void button1_Click_1(object sender, EventArgs e)
        {
            if (pari == true)
            {
                this.pictureBox1.Location = new System.Drawing.Point(400, 162);
                pari = false;
            }
            else {
                this.pictureBox1.Location = new System.Drawing.Point(39, 162);
                pari = true;
            }
        }

        private void button1_Hover(object sender, EventArgs e)
        {
            this.button1.BackColor = Color.FromArgb(r.Next(0, 255), r.Next(0, 255), r.Next(0, 255));
        }

        private void button2_Click(object sender, EventArgs e)
        {
            this.button1.BackColor = Color.FromArgb(r.Next(0, 255), r.Next(0, 255), r.Next(0, 255));
            this.button2.BackColor = Color.FromArgb(r.Next(0, 255), r.Next(0, 255), r.Next(0, 255));
            this.effect1.BackColor = Color.FromArgb(r.Next(0, 255), r.Next(0, 255), r.Next(0, 255));
        }
    }
}
